### PR TITLE
don't merge empty files

### LINKED ISSUE #3118: Do not ask for merging auto-generated datastore-indexes.xml if it is empty
```xml
<!-- Indices written at Wed, 30 May 2018 13:33:43 EDT -->

<datastore-indexes/>
```

We should suppress the pop-up when no index was actually generated like the above.

### PR DESCRIPTION
**Pull request by elharo**
fix #3118 @chanseokoh @briandealwis

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateData.java [hunk 1]
```diff
@@ -49,13 +55,14 @@ static DatastoreIndexUpdateData detect(ILaunchConfiguration configuration, IServ
     LocalAppEngineServerBehaviour serverBehaviour = (LocalAppEngineServerBehaviour) server
         .loadAdapter(LocalAppEngineServerBehaviour.class, null);
     IPath deployPath = serverBehaviour.getModuleDeployDirectory(defaultService);
-    IPath datastoreIndexesAutoXml = deployPath.append("WEB-INF/appengine-generated/datastore-indexes-auto.xml");
-    if(!datastoreIndexesAutoXml.toFile().exists()) {
+    IPath datastoreIndexesAutoXml =
+        deployPath.append("WEB-INF/appengine-generated/datastore-indexes-auto.xml");
+    if(!indexGenerated(datastoreIndexesAutoXml)) {
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateData.java]: `if (`.

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateDataTest.java [hunk 2]
```diff
@@ -148,5 +161,29 @@ private void createDatastoreIndexesAutoXml() throws IOException {
     File datastoreIndexesAutoXml = new File(appengineGenerated, "datastore-indexes-auto.xml");
     assertTrue(appengineGenerated.mkdirs());
     assertTrue(datastoreIndexesAutoXml.createNewFile());
+    
+    try (Writer out = new OutputStreamWriter(new FileOutputStream(datastoreIndexesAutoXml),
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateDataTest.java]: I remember someone told me to prefer `Files.newOutputStream()`.

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateData.java (hunk 1 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateData.java]: done

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateDataTest.java (hunk 2 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/DatastoreIndexUpdateDataTest.java]: done

**Pull request merged**